### PR TITLE
Move internal changes

### PR DESCRIPTION
**Pull request by christopherbate**
[mlir-tensorrt] Move internal changes

This change is a combination of the following:

### [executor] Support lowering `linalg.fill` to runtime memset builtins

Adds support for lowering `linalg.fill` when the following conditions are
true:

1) the element type has bitwidth <= 32
2) the memref is in 'host' or 'device' memory space
3) the memref is contiguous

### NFC: [executor] fix test config for executor integration tests

Executor integration tests were not being run due to incorrect Executor
LIT configuration. This change fixes the config and fixes one test that
was in the incorrect place (it required `cuda-to-executor`, which is not
part of the Executor sub project).

In addition, this change makes some slight modifications to the end-to-end
tests which use the Executor testing bufferization pipeline. We change the
pipeline to run the inliner after bufferization occurs. This in turn requires
fixing the memory space annotations on `memref`` types used in those tests.

### [StableHloToTensorRT] Extending the conversion from `stablehlo.scatter` that conducts the slice update to `tensorrt.scatter_elements`

This entends the existing pass that converts stablehlo.scatter that conducts
the slice update to tensorrt.scatter_elements, adding the support for slicing
operation in axis=0. An example case that this extension covers is:
```
import jax
import jax.numpy as jnp

@jax.jit
def main(
    input_mask, time_step=0, seq_len=1024,
):
  input_mask = (
      jnp.zeros((1, 1024), dtype=jnp.bool_)
      .at[:, :134]
      .set(input_mask)
  )
  return input_mask

if __name__ == "__main__":
    input_mask = jnp.full((1,134), True, dtype=jnp.bool)
    main(input_mask)
```

In our compilation pipeline, this becomes in IRs
```
func.func @main(%arg0: tensor<1x134xi1>) -> tensor<1x1024xi1> {
  %c = stablehlo.constant dense<0> : tensor<1x1xi32>
  %c_0 = stablehlo.constant dense<false> : tensor<1024x1xi1>
  %0 = stablehlo.reshape %arg0 : (tensor<1x134xi1>) -> tensor<1x134x1xi1>
  %1 = "stablehlo.scatter"(%c_0, %c, %0) <{indices_are_sorted = false, scatter_dimension_numbers = #stablehlo.scatter<update_window_dims = [1, 2], scatter_dims_to_operand_dims = [0],
index_vector_dim = 1>, unique_indices = false}> ({
  ^bb0(%arg1: tensor<i1>, %arg2: tensor<i1>):
    stablehlo.return %arg2 : tensor<i1>
  }) : (tensor<1024x1xi1>, tensor<1x1xi32>, tensor<1x134x1xi1>) -> tensor<1024x1xi1>
  %2 = stablehlo.reshape %1 : (tensor<1024x1xi1>) -> tensor<1x1024xi1>
  return %2 : tensor<1x1024xi1>
}
```

### [StableHloExt] Extend SimplifyReshapeBroadcastInDimReshape

The exisiting pattern, SimplifyReshapeBroadcastInDimReshape is extended to cover the case
where the target block's incoming tensor rank is larger than the outgoing tensor rank.
For example, in the following,
```
func.func @main(%arg0: tensor<1x1x1x1x256xf16>) -> tensor<1x1x8x256xf16> {
    %0 = stablehlo.reshape %arg0 : (tensor<1x1x1x1x256xf16>) -> tensor<1x1x1x1x1x1x1x1x256xf16>
    %1 = stablehlo.broadcast_in_dim %0, dims = [0, 1, 2, 3, 4, 5, 7, 8, 9] : (tensor<1x1x1x1x1x1x1x1x256xf16>) -> tensor<1x1x1x1x1x1x8x1x1x256xf16>
    %2 = stablehlo.reshape %1 : (tensor<1x1x1x1x1x1x8x1x1x256xf16>) -> tensor<1x1x8x256xf16>
    return %2: tensor<1x1x8x256xf16>
}
```
tensor<1x1x1x1x256xf16>) has a higer rank than tensor<1x1x8x256xf16>.
In such a case, we first reshape the incoming tensor to match the outgoing tensor, if possible.
As a result, the example above is changed to:
```
func.func @main(%arg0: tensor<1x1x1x1x256xf16>) -> tensor<1x1x8x256xf16> {
    %0 = stablehlo.reshape %arg0 : (tensor<1x1x1x1x256xf16>) -> tensor<1x1x1x256xf16>
    %1 = stablehlo.broadcast_in_dim %0, dims = [0, 1, 2, 3] : (tensor<1x1x1x256xf16>) -> tensor<1x1x8x256xf16>
    return %1 : tensor<1x1x8x256xf16>
```

### [NFC] Fix minor issues

This MR fixes following issues
1. Add license information to `TensorRTVersionCompatibility.cpp`
module.
2. Remove a unnecessary header from `StablehloClustering.cpp`
module
3. Plan dialect pipeline creation was using hardcoded TRT
version i.e. 10. Repace that with `NV_TENSORRT_MAJOR`
to use loaded version.

### [executor] Don't require 'target' attribute for outlining clusters

### CODE REVIEW

**Comment by christopherbate** [comment]
Requires updating the LLVM commit

**Review by shelkesagar29 — APPROVED** [review]
LGTM
